src/db.c

```diff
@@ -837,6 +837,11 @@ void keysCommand(client *c) {
         }
         kvs_di = kvstoreGetDictSafeIterator(c->db->keys, pslot);
     } else {
+        if (!kvstoreSize(c->db->keys)) {
+            /* Requested db is empty. */
+            setDeferredArrayLen(c, replylen, 0);
```

Comment:
Would it make sense to refactor the code like this for clarity ?

```
if ((pslot != -1 && !kvstoreDictSize(c->db->keys, pslot)) || 
    (pslot == -1 && !kvstoreSize(c->db->keys))) {
    // Either the requested slot or db is empty.
    setDeferredArrayLen(c, replylen, 0);
    return;
}

if (pslot != -1) {
    kvs_di = kvstoreGetDictSafeIterator(c->db->keys, pslot);
} else {
    kvs_it = kvstoreIteratorInit(c->db->keys);
}
```

Reply:
Honestly I would rather refactor this to just:
```
if (pslot != -1) {
    kvs_di = kvstoreGetDictSafeIterator(c->db->keys, pslot);
} else {
    kvs_it = kvstoreIteratorInit(c->db->keys);
}
```

and just let it fall through. I don't think we rally care about optimizing for the empty DB case, which should be uncommon. Adding branches hurts branch prediction.

Reply:
ok, good point about the branch prediction, i am not sure about the impact, but let's do it, i removed the branches.